LDACheck_key_phrases.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Istex_Mental_Rotation.
+# Copyright (C) 2016 3ST ERIC Laboratory.
+#
+# This is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+# Load the SVD representation of documents done for the whole corpus of ISTEX and UCBL.
+# Classify the documents by clusters using the LatentDirichletAllocation method. Try with different number of clusters.
+# Extract the key words representing each cluster.
+
+# co-author : Lucie Martinet <lucie.martinet@univ-lorraine.fr>
+# co-author : Hussein AL-NATSHEH <hussein.al-natsheh@ish-lyon.cnrs.fr.>
+# Affiliation: University of Lyon, ERIC Laboratory, Lyon2
+
+# Thanks to ISTEX project for the foundings
+
+import os, argparse, pickle, json
+from sklearn.feature_extraction.text import CountVectorizer,  TfidfVectorizer
+from sklearn.metrics.pairwise import cosine_similarity
+from sklearn.decomposition import TruncatedSVD
+
+import IPython
+from sklearn.decomposition import LatentDirichletAllocation
+import numpy as np
+import sys
+reload(sys)
+sys.setdefaultencoding('utf8')
+
+def print_top_words(model, feature_names, n_top_words):
+	for topic_idx, topic in enumerate(model.components_):
+		print("Topic #%d:" % topic_idx)
+		print(" | ".join([feature_names[i]
+		for i in topic.argsort()[:-n_top_words - 1:-1]]))
+	print()
+
+def write_top_words(model, feature_names, n_top_words, outfile):
+	for topic_idx, topic in enumerate(model.components_):
+		outfile.write("Topic #%d:" % topic_idx)
+		outfile.write("\n")
+		outfile.write(" | ".join([feature_names[i]
+		for i in topic.argsort()[:-n_top_words - 1:-1]]))
+		outfile.write("\n")
+	outfile.write("\n")
+
+def KeysValuesInit(nb_articles) : 
+	keys = np.array(range(nb_articles),dtype=np.object)
+	values = np.array(range(nb_articles),dtype=np.object)
+	for i, (key,value) in enumerate(input_dict.items()) :
+		keys[i] = key
+		values[i] = value
+
+	return keys, values
+
+def statisticsClusterSelection(cluster, document_id, docs_topic, selection, stat_selection, outfile_pointer):
+	if selection in document_id and outfile_pointer != None and len(selection)==len(document_id.split("_")[0]): # keys[t] is a string, the name of the document
```

Comment:
in the documentation, what does 'keys[t]' refer to in the function code? Could you please also break this line into 2 lines (one for the code and the other for the documentation)? Same for the line after